### PR TITLE
Use flask-base

### PR DESCRIPTION
**Pull request by jkfran**
## Done

Changes to use flask-base

## QA

- Pull the branch
- Run the site using the command `./run`
- View the site locally in your web browser at: http://0.0.0.0:8004/
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)

### CODE REVIEW

File: webapp/app.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 The web frontend for the snap store.
 """
 
-import flask
+from canonicalwebteam.flask_base.app import FlaskBase
 from werkzeug.contrib.fixers import ProxyFix
```
Comment (nottrobin) [inline webapp/app.py]: This should be removed - it's included in flask-base:

https://github.com/canonical-web-and-design/canonicalwebteam.flask-base/blob/master/canonicalwebteam/flask_base/app.py#L58

File: webapp/app.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
...
 from werkzeug.debug import DebuggedApplication
```
Comment (nottrobin) [inline webapp/app.py]: This too: https://github.com/canonical-web-and-design/canonicalwebteam.flask-base/blob/master/canonicalwebteam/flask_base/app.py#L56

File: requirements.txt [hunk 3]
```diff
@@ -7,7 +7,7 @@ canonicalwebteam.yaml-responses==1.1.1
 canonicalwebteam.image-template==1.0.0
```
Comment (nottrobin) [inline requirements.txt]: yaml-responses on the previous line should also be removed as it's included in flask-base:

https://github.com/canonical-web-and-design/canonicalwebteam.flask-base/blob/master/canonicalwebteam/flask_base/app.py#L62

Comment (jkfran) [reply requirements.txt]: Done!

File: webapp/app.py [hunk 4]
```diff
@@ -4,19 +4,13 @@
 The web frontend for the snap store.
 """
 
-import flask
-from werkzeug.contrib.fixers import ProxyFix
-from werkzeug.debug import DebuggedApplication
+from canonicalwebteam.flask_base.app import FlaskBase
 
 import talisker.flask
```
Comment (nottrobin) [inline webapp/app.py]: Sorry I didn't catch this in the previous review round, but these 2 lines should also be removed as they're in flask-base:

https://github.com/canonical-web-and-design/canonicalwebteam.flask-base/blob/master/canonicalwebteam/flask_base/app.py#L85

**Comment by nottrobin** [comment]
You should also make use of `favicon_url` to replace `webapp/snapcraft/views.py` line 247

File: webapp/app.py [hunk 5]
```diff
@@ -33,17 +27,16 @@
...
-
     app.config.from_object("webapp.config")
     app.testing = testing
 
-    app.wsgi_app = ProxyFix(app.wsgi_app)
-    if app.debug:
-        app.wsgi_app = DebuggedApplication(app.wsgi_app)
-
     app.url_map.strict_slashes = False
     app.url_map.converters["regex"] = helpers.RegexConverter
```
Comment (nottrobin) [inline webapp/app.py]: These 2 lines are also included in flask-base:

https://github.com/canonical-web-and-design/canonicalwebteam.flask-base/blob/master/canonicalwebteam/flask_base/app.py#L53

File: webapp/app.py [hunk 6]
```diff
@@ -56,10 +49,6 @@ def create_app(testing=False):
         talisker.logs.set_global_extra({"service": "snapcraft.io"})
```
Comment (nottrobin) [inline webapp/app.py]: The two lines above here, where we configure two separate sessions with talisker.... this should be done differently but I guess this is out of scope of this PR.

Ideally we would have a single session that we create here with `talisker.requests.get_session()` and then we pass through to these API libraries.

But this is out of scope from this PR, so we should probably create an issue about it.

**Comment by nottrobin** [comment]
Oh you should also remove the `robots.txt` and `humans.txt` logic from snapcraft.io in favour of what's in flask-base:

https://github.com/canonical-web-and-design/canonicalwebteam.flask-base/blob/master/canonicalwebteam/flask_base/app.py#L115

It looks like snapcraft sends an empty response for `/robots.txt` - I think a more straightforward solution is to use flask-base's default behaviour and create an empty file at `/static/robots.txt`.

File: webapp/app.py (hunk 5 above)
Comment (jkfran) [reply webapp/app.py]: I get an error related with this, Werkzeug is [used in Flask-base here](https://github.com/canonical-web-and-design/canonicalwebteam.flask-base/blob/master/canonicalwebteam/flask_base/converters.py#L1) but it is not a dependency in the library:
```
  File "/home/fran/Workspace/snapcraft.io/webapp/app.py", line 9, in <module>
    from canonicalwebteam.flask_base.app import FlaskBase
  File "/usr/local/lib/python3.6/dist-packages/canonicalwebteam/flask_base/app.py", line 7, in <module>
    from werkzeug.middleware.proxy_fix import ProxyFix
ModuleNotFoundError: No module named 'werkzeug.middleware'
```

Comment (jkfran) [reply webapp/app.py]: Solved!

**Comment by jkfran** [comment]
@nottrobin Can you review this again with all these changes? Also, for the `robots.txt` and `humans.txt` I got a 404, these files are expected to be in the root folder:
https://github.com/canonical-web-and-design/canonicalwebteam.flask-base/blob/master/canonicalwebteam/flask_base/app.py#L112-L113

I moved both files to the root folder if that is fine.
